status-group-chats-spec.md

```diff
@@ -28,177 +28,105 @@ This documents describes the group chat protocol used by the status application.
 
 ## Membership updates
 
-Membership updates messages are used to propagate group chat membership changes. The transit format is described in the [Status Payload Specs](status-payload-specs.md). Here we will be describing each specific field.
-
-The format is:
+Membership updates messages are used to propagate group chat membership changes. The protobuf format is described in the [Status Payload Specs](status-payload-specs.md). Here we will be describing each specific field.
+
+The protobuf messages are:
+
+```protobuf
+// MembershipUpdateMessage is a message used to propagate information
+// about group membership changes.
+message MembershipUpdateMessage {
+  // The chat id of the private group chat
+  string chat_id = 1;
+  // A list of events for this group chat, first x bytes are the signature, then is a 
+  // protobuf encoded MembershipUpdateEvent
+  repeated bytes events = 2;
+  // An optional chat message
+  ChatMessage message = 3;
+}
 
-```
-{
-  "events": [struct {"type": string, "member": string, "members": [string], "clock-value": uint, "name": string],
-  "signature": string,
-  "chat-id": string
+message MembershipUpdateEvent {
+  // Lamport timestamp of the event
+  uint64 clock = 1;
+  // List of public keys of objects of the action
+  repeated string members = 2;
+  // Name of the chat for the CHAT_CREATED/NAME_CHANGED event types
+  string name = 3;
+  // The type of the event
+  EventType type = 4;
+
+  enum EventType {
+    UNKNOWN = 0;
+    CHAT_CREATED = 1;
+    NAME_CHANGED = 2;
+    MEMBERS_ADDED = 3;
+    MEMBER_JOINED = 4;
+    MEMBER_REMOVED = 5;
+    ADMINS_ADDED = 6;
+    ADMIN_REMOVED = 7;
+  }
 }
 ```
 
 ### Chat ID
 
-Each membership update MUST be sent with a corresponding `chat-id`. 
-The format of this chat id MUST be a string, [UUID](https://tools.ietf.org/html/rfc4122 ), concatenated with the hex-encoded public key of the creator of the chat. This chat-id MUST be validated by all clients, and MUST be discarded if it does not follow these rules.
+Each membership update MUST be sent with a corresponding `chatId`. 
+The format of this chat id MUST be a string, [UUID](https://tools.ietf.org/html/rfc4122 ), concatenated with the hex-encoded public key of the creator of the chat. This chatId MUST be validated by all clients, and MUST be discarded if it does not follow these rules.
```

Comment:
```suggestion
The format of this chat ID MUST be a string of [UUID](https://tools.ietf.org/html/rfc4122) concatenated with the hex-encoded public key of the creator of the chat. This chat ID MUST be validated by all clients, and MUST be discarded if it does not follow these rules.
```

Do we use `-` for concatenation or nothing?

Reply:
we do, updated